projects/ADuCM360_demo_cn0409/src/ADPD105.c

```diff
@@ -0,0 +1,487 @@
+
+	u16Value = ADPD105_ReadReg(ADPD105_PD_LED_SELECT);
+	u16Value &= 0xF00F; //reset before settings for PD Time slot
+	u16Value |= ((uint16_t)u8SlotASelect << 4) | (uint16_t)(u8SlotBSelect << 8); //set time slot values PD
+
+	ADPD105_WriteReg(ADPD105_PD_LED_SELECT, u16Value);
+}
+
+
+/**
+   @brief Enable the internal 32kHz internal clock from ADPD105 via I2C.
+
+   @param u8SlotBSelect - u8Enable - enables/disables 32kHz clock
+
+   @return void
+
+**/
+void ADPD105_SetCLK32K(uint8_t u8Enable)
+{
+	uint16_t u16Value;
+
+	u16Value = ADPD105_ReadReg(ADPD105_SAMPLE_CLK);
+	ADPD105_WriteReg(ADPD105_SAMPLE_CLK, u16Value | (u8Enable << 7)); //enable 32kHz internal clock
+}
+
+/**
+   @brief Setup FIFO for data reading from ADPD105.
+
+   @return void
+
+**/
+void ADPD105_SetFIFO(void)
+{
+	ADPD105_SetOperationMode(PROGRAM); //set programming mode
+
+	ADPD105_WriteReg(ADPD105_SLOT_EN, 0x3131); //Slot A enable and 32 bit extend sample data to fifo
+	ADPD105_WriteReg(ADPD105_FIFO_THRESH, 0x1F00); //set FIFO threshold to 8 words data
+	ADPD105_WriteReg(ADPD105_INT_MASK, 0x0FF); //Set FIFO interrupt
+	ADPD105_WriteReg(ADPD105_GPIO_DRV, 0x05); //set GPIO
+
+	ADPD105_SetOperationMode(NORMAL_OPERATION); //enter normal operation
+
+}
+
+/**
+   @brief Select what led to be used inside a time slot.
+
+   @param enLEDNumber - the led number, see ADPD105_LED
+
+   @param enSlot - time slot ( SlotA or SlotB)
+
+   @return void
+
+**/
+void ADPD105_SelectLED(ADPD105_LED enLEDNumber, ADPD105_TimeSlot enSlot)
+{
+	uint16_t u16Value;
+
+	u16Value = ADPD105_ReadReg(ADPD105_PD_LED_SELECT); //read register value
+
+	if (enSlot == SLOTA) {
+		u16Value &= 0xFFFC;
+		u16Value |= enLEDNumber; //configure LED for SLOTA
+	} else {
+		u16Value &= 0xFFF3;
+		u16Value |= (enLEDNumber << 2); //configure LED for SLOTB
+	}
+
+	ADPD105_WriteReg(ADPD105_PD_LED_SELECT, u16Value); //write value to register
+}
+
+/**
+   @brief Deselect the leds for each time slot .
+
+   @return void
+
+**/
+void ADPD105_DeselectLEDs(void)
+{
+	uint16_t u16Value;
+
+	u16Value = ADPD105_ReadReg(ADPD105_PD_LED_SELECT); //read register value
+	u16Value &= 0xFFF0;
+	ADPD105_WriteReg(ADPD105_PD_LED_SELECT, u16Value); //write value to register
+}
+
+
+/**
+   @brief Read the data register for each time slot
+
+   @return void
+
+**/
+void ADPD105_ReadDataRegs(void)
+{
+
+	timer_sleep(30);
+
+	/*Set data hold - disable data update*/
+	ADPD105_WriteReg(ADPD105_DATA_ACCESS_CTL, 0x07);
+
+	/*Read data registers*/
+	au16DataSlotA[0] = ADPD105_ReadReg(ADPD105_SLOTA_PD1_16BIT);
+	au16DataSlotA[1] = ADPD105_ReadReg(ADPD105_SLOTA_PD2_16BIT);
+	au16DataSlotA[2] = ADPD105_ReadReg(ADPD105_SLOTA_PD3_16BIT);
+	au16DataSlotA[3] = ADPD105_ReadReg(ADPD105_SLOTA_PD4_16BIT);
+
+	au16DataSlotB[0] = ADPD105_ReadReg(ADPD105_SLOTB_PD1_16BIT);
+	au16DataSlotB[1] = ADPD105_ReadReg(ADPD105_SLOTB_PD2_16BIT);
+	au16DataSlotB[2] = ADPD105_ReadReg(ADPD105_SLOTB_PD3_16BIT);
+	au16DataSlotB[3] = ADPD105_ReadReg(ADPD105_SLOTB_PD4_16BIT);
```

Comment:
I don't like the fact that this function uses a global array; maybe pass a pointer to this array as a parameter to this function?